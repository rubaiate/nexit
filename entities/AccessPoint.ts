import { Entity, Column, PrimaryGeneratedColumn } from 'typeorm/browser'

@Entity('access_points')
export default class AccessPoint {
    @PrimaryGeneratedColumn()
    id!: number;

    @Column('int')
    station_id: number

    @Column("varchar")
    type!: string;

    @Column('int')
    base_gate: number = 0;

    @Column('int')
    base_gate_align: number = 0;

    @Column('int')
    ap_align: number = 0;

    @Column('int')
    access_direction: number = 0;

    @Column('int')
    x_offset: number = 0;

    @Column('int')
    y: number = 0;

}