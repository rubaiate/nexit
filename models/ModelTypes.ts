export enum Alignment {
    Start,
    End,
    Mid
}

export interface TrainLine {
    name:string,
    prefix:string,
    gateCount: number,
    stationCount: number,
}

export interface AccessPoint {
    type:string,
    baseGate:number,
    baseGateAlign:Alignment,
    apAlign:Alignment,
    xOffset:number,
    y:number,
    accessDirection:number,
}

export interface Station {
    trainLine:TrainLine,
    name:string,
    code:string,
    lat:number,
    lon:number,
    stationIndex:number
    accessPoints:AccessPoint[]
}

