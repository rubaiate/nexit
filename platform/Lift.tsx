import * as React from "react";
import Svg, { Path, G } from "react-native-svg";
import RoundedRecatSvg from "../components/RoundedRectSvg";
interface Props {
  x: number;
  y: number;
  width: number;
  height: number;
}

function SvgComponent(props: Props) {
  const { x, y, width, height } = props;
  return (
    <G x={x} y={y}>
      <RoundedRecatSvg {...props} r={10} stroke="black" fill="white" />
      <Svg
        viewBox="0 0 297.98 487.59"
        width={width}
        height={height}
        fill="black"
        stroke="black"
      >
        <Path d="M0 448.04v-218l.046-.01c0-21.74 16.721-39.8 39.721-39.8h218.5c22.25 0 39.7 17.81 39.7 39.81v218l.01.05c.04 23.2-17.71 39.49-39.46 39.49H39.767l-.003.01c-21.75 0-39.767-17.3-39.767-39.55z" />
        <Path d="M41.174 231.9h215.62v214.37H41.174l.004-214.37z" fill="#fff" />
        <Path d="M202.89 21.506c-.24-28.514 41.01-28.836 41.25 0l-.62 86.524 22.18-27.179c15.32-16.813 42.26 6.625 25.63 25.629l-53.81 64.06c-8.5 8.5-19.55 8.45-28 0l-53.57-64.31c-16.62-19.004 10.57-42.692 25.63-25.629l21.94 27.429-.63-86.524zM53.142 155.57c-.24 28.51 41.01 28.84 41.25 0l-.627-86.526 22.185 27.182c15.32 16.814 42.25-6.625 25.63-25.625L87.767 6.538c-8.5-8.5-19.55-8.449-28 0L6.205 70.851c-16.626 19 10.562 42.69 25.624 25.625l21.94-27.432-.627 86.526z" />
        <G>
          <Path d="M246.57 354.7c0 4.34-3.49 7.36-7.33 7.36h-48.29c-3.79 0-7.34-3.07-7.34-7.36v-52.47c0-6.02 4.82-10.14 10.09-10.14h42.75c5.75 0 10.12 4.57 10.12 10.14v52.47z" />
          <Path d="M234.77 360.14H216.4v70.96c0 12.59 18.37 12.59 18.37 0v-70.96zM213.6 360.14h-18.37v70.96c0 12.49 18.37 12.49 18.37 0v-70.96z" />
          <G>
            <Path d="M215.27 287.83c7.61 0 13.79-6.18 13.79-13.79 0-7.62-6.18-13.79-13.79-13.79-7.62 0-13.79 6.17-13.79 13.79 0 7.61 6.17 13.79 13.79 13.79z" />
          </G>
          <G>
            <Path d="M180.57 354.7c0 4.34-3.49 7.36-7.33 7.36h-48.29c-3.79 0-7.34-3.07-7.34-7.36v-52.47c0-6.02 4.82-10.14 10.09-10.14h42.75c5.75 0 10.12 4.57 10.12 10.14v52.47z" />
            <Path d="M168.77 360.14H150.4v70.96c0 12.59 18.37 12.59 18.37 0v-70.96zM147.6 360.14h-18.37v70.96c0 12.49 18.37 12.49 18.37 0v-70.96z" />
            <G>
              <Path d="M149.27 287.83c7.61 0 13.79-6.18 13.79-13.79 0-7.62-6.18-13.79-13.79-13.79-7.62 0-13.79 6.17-13.79 13.79 0 7.61 6.17 13.79 13.79 13.79z" />
            </G>
          </G>
          <G>
            <Path d="M114.57 354.7c0 4.34-3.49 7.36-7.33 7.36H58.954c-3.791 0-7.344-3.07-7.344-7.36v-52.47c0-6.02 4.824-10.14 10.094-10.14h42.746c5.75 0 10.12 4.57 10.12 10.14v52.47z" />
            <Path d="M102.77 360.14H84.394v70.96c0 12.59 18.376 12.59 18.376 0v-70.96zM81.602 360.14H63.23v70.96c0 12.49 18.372 12.49 18.372 0v-70.96z" />
            <G>
              <Path d="M83.267 287.83c7.615 0 13.789-6.18 13.789-13.79 0-7.62-6.174-13.79-13.789-13.79s-13.789 6.17-13.789 13.79c0 7.61 6.174 13.79 13.789 13.79z" />
            </G>
          </G>
        </G>
      </Svg>
    </G>
  );
}

export default SvgComponent;
