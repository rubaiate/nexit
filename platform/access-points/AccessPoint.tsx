import { ReactElement, ReactNode } from "react";
import { AccessPointProps } from "./AccessPointTypes";
import React from "react";

import Escalator from "./Escalator";
import Lift from "./Lift";
import Staircase from "./Staircase";
import { G } from "react-native-svg";
import { Alignment } from "../../models/ModelTypes";

const accessPoints: { [key: string]: any } = {
  escalator: Escalator,
  lift: Lift,
  staircase: Staircase,
};

const AccessPoint = (props: AccessPointProps) => {
  const { x, y, width, apAlign, type } = props;
  const Element = accessPoints[type];

  let xPos;
  switch (apAlign) {
    case Alignment.End:
      xPos = x - width;
      break;
    case Alignment.Mid:
      xPos = x - width / 2;
      break;
    default:
      xPos = x;
  }

  if (Element) {
    return (
      <G x={xPos} y={y}>
        <Element {...props} />
      </G>
    );
  }

  return <div>Invalid</div>;
};

export default AccessPoint;
