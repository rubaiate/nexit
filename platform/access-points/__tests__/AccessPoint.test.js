import React from 'react';
import renderer from 'react-test-renderer';
import { G } from "react-native-svg";
import AccessPoint from '../AccessPoint';
import Lift from '../Lift'
import { Alignment } from '../../../models/ModelTypes';

it('render switch should render lift', () => {
    const tree = renderer.create(<AccessPoint x={300} y={250} width={100} height={100} accessDirection={1} type="lift"/>).toJSON();
    const treeLift = renderer.create(<G x={300} y={250}><Lift x={300} y={250} width={100} height={100} accessDirection={1}/></G>).toJSON();
    expect(tree).toEqual(treeLift);
});

it('align by start', () => {
    const tree = renderer.create(<AccessPoint x={300} y={250} width={100} height={100} accessDirection={1} type="staircase" apAlign={Alignment.Start}/>).toJSON();
    expect(tree).toMatchSnapshot()
});

it('align by end', () => {
    const tree = renderer.create(<AccessPoint x={300} y={250} width={100} height={100} accessDirection={1} type="staircase" apAlign={Alignment.End}/>).toJSON();
    expect(tree).toMatchSnapshot()
});

it('align by mid', () => {
    const tree = renderer.create(<AccessPoint x={300} y={250} width={100} height={100} accessDirection={1} type="staircase" apAlign={Alignment.Mid}/>).toJSON();
    expect(tree).toMatchSnapshot()
});