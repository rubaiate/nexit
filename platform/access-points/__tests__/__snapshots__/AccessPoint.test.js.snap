// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`align by end 1`] = `
<RNSVGGroup
  matrix={
    Array [
      1,
      0,
      -0,
      1,
      200,
      250,
    ]
  }
>
  <RNSVGPath
    d="M6,-4 h88 a10,10 0 0 1 10,10 v88 a10,10 0 0 1 -10,10 h-88 a10,10 0 0 1 -10,-10 v-88 a10,10 0 0 1 10,-10"
    fill={4294967295}
    propList={
      Array [
        "fill",
        "stroke",
      ]
    }
    stroke={4278190080}
  />
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={100}
    bbWidth={100}
    fill="black"
    focusable={false}
    height={100}
    meetOrSlice={0}
    minX={0}
    minY={0}
    stroke="black"
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        Object {
          "flex": 0,
          "height": 100,
          "width": 100,
        },
      ]
    }
    vbHeight={512.004}
    vbWidth={512.004}
    width={100}
  >
    <RNSVGGroup
      fill={4278190080}
      propList={
        Array [
          "fill",
          "stroke",
        ]
      }
      stroke={4278190080}
    >
      <RNSVGPath
        d="M364.957 107.108v34.861h15.057v62.005H268.09v34.861h15.058v62.006H171.223v34.862h15.059v62.005H74.356v34.861h15.059v79.435h34.861v-79.435h96.867v-96.866h96.866v-96.868h96.867v-96.866h79.436v-34.861zM17.69 321.282L338.97.002l24.651 24.65-321.28 321.28z"
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</RNSVGGroup>
`;

exports[`align by mid 1`] = `
<RNSVGGroup
  matrix={
    Array [
      1,
      0,
      -0,
      1,
      250,
      250,
    ]
  }
>
  <RNSVGPath
    d="M6,-4 h88 a10,10 0 0 1 10,10 v88 a10,10 0 0 1 -10,10 h-88 a10,10 0 0 1 -10,-10 v-88 a10,10 0 0 1 10,-10"
    fill={4294967295}
    propList={
      Array [
        "fill",
        "stroke",
      ]
    }
    stroke={4278190080}
  />
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={100}
    bbWidth={100}
    fill="black"
    focusable={false}
    height={100}
    meetOrSlice={0}
    minX={0}
    minY={0}
    stroke="black"
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        Object {
          "flex": 0,
          "height": 100,
          "width": 100,
        },
      ]
    }
    vbHeight={512.004}
    vbWidth={512.004}
    width={100}
  >
    <RNSVGGroup
      fill={4278190080}
      propList={
        Array [
          "fill",
          "stroke",
        ]
      }
      stroke={4278190080}
    >
      <RNSVGPath
        d="M364.957 107.108v34.861h15.057v62.005H268.09v34.861h15.058v62.006H171.223v34.862h15.059v62.005H74.356v34.861h15.059v79.435h34.861v-79.435h96.867v-96.866h96.866v-96.868h96.867v-96.866h79.436v-34.861zM17.69 321.282L338.97.002l24.651 24.65-321.28 321.28z"
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</RNSVGGroup>
`;

exports[`align by start 1`] = `
<RNSVGGroup
  matrix={
    Array [
      1,
      0,
      -0,
      1,
      300,
      250,
    ]
  }
>
  <RNSVGPath
    d="M6,-4 h88 a10,10 0 0 1 10,10 v88 a10,10 0 0 1 -10,10 h-88 a10,10 0 0 1 -10,-10 v-88 a10,10 0 0 1 10,-10"
    fill={4294967295}
    propList={
      Array [
        "fill",
        "stroke",
      ]
    }
    stroke={4278190080}
  />
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={100}
    bbWidth={100}
    fill="black"
    focusable={false}
    height={100}
    meetOrSlice={0}
    minX={0}
    minY={0}
    stroke="black"
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        Object {
          "flex": 0,
          "height": 100,
          "width": 100,
        },
      ]
    }
    vbHeight={512.004}
    vbWidth={512.004}
    width={100}
  >
    <RNSVGGroup
      fill={4278190080}
      propList={
        Array [
          "fill",
          "stroke",
        ]
      }
      stroke={4278190080}
    >
      <RNSVGPath
        d="M364.957 107.108v34.861h15.057v62.005H268.09v34.861h15.058v62.006H171.223v34.862h15.059v62.005H74.356v34.861h15.059v79.435h34.861v-79.435h96.867v-96.866h96.866v-96.868h96.867v-96.866h79.436v-34.861zM17.69 321.282L338.97.002l24.651 24.65-321.28 321.28z"
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</RNSVGGroup>
`;
